### ISSUE TITLE
Unicode in features

### ISSUE THREAD

**Issue by memee** [behave/behave#134]
```
Traceback (most recent call last):
  File "/Users/mmetelko/.virtualenvs/sor_interface/bin/behave", line 9, in <module>
    load_entry_point('behave==1.2.2', 'console_scripts', 'behave')()
  File "/Users/mmetelko/.virtualenvs/sor_interface/lib/python2.7/site-packages/behave/__main__.py", line 120, in main
    sys.stderr.write(escapes['undefined'] + msg + escapes['reset'])
UnicodeEncodeError: 'ascii' codec can't encode characters in position 92-93: ordinal not in range(128)
```

**Comment by bittner** [comment]
Hi, I'm following up on this issue, because I'm currently running into what I believe is the same error message:
``` bash
$ behave --version
behave 1.2.5
$ behave --lang-list
Languages available:
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/myproject/bin/behave", line 11, in <module>
    sys.exit(main())
  File "/home/vagrant/.virtualenvs/myproject/local/lib/python2.7/site-packages/behave/__main__.py", line 70, in main
    print(u'%s: %s / %s' % (iso_code, native, name))
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 4-10: ordinal not in range(256)
```

## System setup and configuration

Essentially, I have a fresh installation of behave (version **1.2.5**) with Python **2.7.3**, and I'm trying to list the available languages. That's it.

Here is what I have in my setup.cfg file: (though I don't think it matters)
``` ini
[behave]
junit = yes
paths = tests
summary = yes
lang = de
```
I'm using Ubuntu 12.04 Precise (as a Vagrant box on a Mac OS X 10.10.3 Yosemite system), hence the aged Python version.

## Motivation for listing languages (lateral problem)

What I was trying to do, actually, is print some help on using other languages than English. Because I want behave to print German umlauts as correct unicode characters in the step snippets generated from a fresh `.feature` file. They are currently printed to the terminal as question marks, e.g.
``` python
@then(u'die Checkbox "Ich m?chte eingeloggt bleiben" ist abgew?hlt')
def step_impl(context):
    raise NotImplementedError(u'STEP: Then die Checkbox "Ich m?chte eingeloggt bleiben" ist abgew?hlt')
``` 
I didn't find any information in the docs on forcing behave to use a codec when interpreting the `.feature` file.

## Analysis (tl;dr Mac :+1:, Ubuntu VM :-1:)

I've noticed that listing the languages works on the Mac (not the Linux system), which runs Python **2.7.6**, btw. Also, umlauts and other foreign (non-ASCII) characters are printed as gibberish on Ubuntu (in the Mac Terminal) but printed fine on the local (non-Vagrant) installation, for both behave help texts and step snippet output. -- I'll try using an Ubuntu 14.04 Trusty as the VM OS, and see if this changes anything on the issue.

It would be nice to figure out what's causing the problem. We're planning to use behave exclusively from within Vagrant boxes (i.e. Ubuntu machines on Macs) writing most feature tests in German, and this would be kind of a show stopper.

Should we reopen this issue?

**Comment by bittner** [comment]
Unfortunately, Ubuntu 14.04 Trusty gives no better result:
```  bash
$ behave --lang-list
Languages available:
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/myproject/bin/behave", line 11, in <module>
    sys.exit(main())
  File "/home/vagrant/.virtualenvs/myproject/local/lib/python2.7/site-packages/behave/__main__.py", line 70, in main
    print(u'%s: %s / %s' % (iso_code, native, name))
UnicodeEncodeError: 'ascii' codec can't encode characters in position 4-10: ordinal not in range(128)
$ behave --lang-help de
Translations for German / Deutsch
             And: Und
            Then: Dann
Scenario Outline: Szenariogrundriss
             But: Aber
        Examples: Beispiele
      Background: Grundlage
           Given: Angenommen, Gegeben sei
        Scenario: Szenario
            When: Wenn
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/myproject/bin/behave", line 11, in <module>
    sys.exit(main())
  File "/home/vagrant/.virtualenvs/myproject/local/lib/python2.7/site-packages/behave/__main__.py", line 86, in main
    u', '.join(w for w in trans[kw] if w != '*')))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe4' in position 30: ordinal not in range(128)
```
Version used are:
```  bash
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty
$ python --version
Python 2.7.6
$ behave --version
behave 1.2.5
```
This problem does not occur on an DELL notebook running Ubuntu 14.04 Trusty natively. This may be a locale problem, though installing language-packs and regenerating UTF-8 locales doesn't fix anything.

Any thoughts?

**Comment by jenisys** [comment]
@bittner 
In general, when you want to use "Umlaute" use UTF-8 as encoding in the feature-files (and step-files). Feature-files are parsed with encoding=UTF-8. 
An example that is works in principle is given in [Tutorial 12](https://github.com/jenisys/behave.example/blob/master/features/tutorial12_spoken_language.feature) in the [behave.example](https://github.com/jenisys/behave.example/) repository.

The ``behave --lang-list`` problem is already fixed in behave-1.2.6dev in the repo (issue #309).

NOTE:
If there are additional (unsolved) Unicode problems, it will be better to open a new ticket (in the future).
